### PR TITLE
string: optimize starts_with & ends_with

### PR DESCRIPTION
**Pull request by yuyi98**
This PR is to optimize `starts_with` and `ends_with` in module `string`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -196,19 +199,16 @@ this backend.
...
-
-
-
-
 ## V 0.1.18
 *16 Aug 2019*
 
-- Built-in ORM (`uk_customers = db.select from Customer where country == 'uk' && nr_orders > 0`)
-- Map initialization syntax: `m := { ‘foo’: ‘bar’, ‘baz’: ‘foo’ }`
+- Built-in ORM (`uk_customers = db.select from Customer where country == 'uk'
```
Comment (spytheman) [inline CHANGELOG.md]: I think it is better to not split the example code to 2 lines.

Comment (spytheman) [reply CHANGELOG.md]: Instead, the whole:
`uk_customers = db.select from Customer where country == 'uk' && nr_orders > 0`
can be on its own line

**Comment by alexesprit** [comment]
Please, don't include commits from your other PRs. This makes reviewing harder. :)

**Comment by yuyi98** [comment]
Excuse me, I don't know how to exclude other commits in the new PR,
I have created new branch to edit and commit, but the new PR still contain
original commit.

Can you tell me how to do it?

 
From: Alexey
Date: 2020-02-19 01:45
To: vlang/v
CC: yuyi; Author
Subject: Re: [vlang/v] string: optimize starts_with & ends_with (#3770)
Please, don't include commits from your other PRs. This makes reviewing harder. :)

**Comment by alexesprit** [comment]
> I have created new branch to edit and commit, but the new PR still contain
original commit.

You should checkout your master branch, and then create a new feature branch from it. Seems like your workflow was following:
1. You created a new branch-a.
2. You made changes and pushed this branch to remote.
3. You created a new branch-b.
4. You made changes and pushed this branch to remote.

In this case your branch-b contains commits from the branch-a, as it's based on it. You should to checkout the master branch first before creating new ones (except when the branch-b requires changes from the branch-a, but it's another task which is more complicated).

I hope this information will help you. Feel free to ask questions - you can join us in Discord and ask questions in Discord channels, if you haven't do that, of course.

**Comment by yuyi98** [comment]
You're right, I understand my problem.

Thank you very much!

 
From: Alexey
Date: 2020-02-19 17:28
To: vlang/v
CC: yuyi; State change
Subject: Re: [vlang/v] string: optimize starts_with & ends_with (#3770)
I have created new branch to edit and commit, but the new PR still contain
original commit.
You should checkout your master branch, and then create a new feature branch from it. Seems like your workflow was following:
You created a new branch-a.
You made changes and pushed this branch to remote.
You created a new branch-b.
You made changes and pushed this branch to remote.
In this case your branch-b contains commits from the branch-a, as it's based on it. You should to checkout the master branch first before creating new ones (except when the branch-b requires changes from the branch-a, but it's another task which is more complicated).
I hope this information will help you. Feel free to ask questions - you can join us in Discord and ask questions in Discord channels, if you haven't do that, of course.

**Pull request closed without merging**